src/ui/SearchInterface/FacetColumnAutoLayoutAdjustment.ts

```diff
@@ -0,0 +1,97 @@
+import { $$ } from '../../utils/Dom';
+import { Logger } from '../../misc/Logger';
+import { QueryEvents } from '../../events/QueryEvents';
+import { every, find } from 'underscore';
+import { get } from '../Base/RegisteredNamedMethods';
+
+export interface IAutoLayoutAdjustableInsideFacetColumn {
+  isCurrentlyDisplayed: () => boolean;
+}
+
+export class FacetColumnAutoLayoutAdjustment {
+  public static autoLayoutAdjustmentComponent: Map<HTMLElement, IAutoLayoutAdjustableInsideFacetColumn[]> = new Map();
+  public static autoLayoutAdjustmentHandlers: Map<HTMLElement, () => void> = new Map();
+
+  public static isAutoLayoutAdjustable(component: any): component is IAutoLayoutAdjustableInsideFacetColumn {
+    return 'isCurrentlyDisplayed' in component;
+  }
+
+  public static initializeAutoLayoutAdjustment(root: HTMLElement, component: IAutoLayoutAdjustableInsideFacetColumn) {
+    if (!this.autoLayoutAdjustmentComponent.has(root)) {
+      this.autoLayoutAdjustmentComponent.set(root, []);
+    }
+
+    this.autoLayoutAdjustmentComponent.get(root).push(component);
+
+    if (this.autoLayoutAdjustmentHandlers.has(root)) {
+      return;
+    }
+    const hiddenClass = 'coveo-no-visible-facet';
+
+    const handler = () =>
+      $$(root).on(QueryEvents.deferredQuerySuccess, () => {
+        const column = this.findColumn(root);
+        const dropDownHeader = this.findDropdownHeader(root);
+
+        if (this.everyStandardComponentsAreInvisible(root) && this.columnsDoesNotContainVisibleCustomElement(column)) {
+          column ? $$(column).addClass(hiddenClass) : null;
+          dropDownHeader ? $$(dropDownHeader).addClass(hiddenClass) : null;
+        } else {
+          column ? $$(column).removeClass(hiddenClass) : null;
+          dropDownHeader ? $$(dropDownHeader).removeClass(hiddenClass) : null;
+        }
+      });
+
+    handler();
+
+    this.autoLayoutAdjustmentHandlers.set(root, handler);
+  }
+
+  private static columnsDoesNotContainVisibleCustomElement(column: HTMLElement) {
+    if (!column) {
+      return true;
+    }
+
+    const childrens = $$(column).children();
+    const shouldBeIgnored = ['coveo-facet-header-filter-by-container', 'coveo-topSpace', 'coveo-bottomSpace'];
```

Comment:
Would it be possible to have some kind of tests that makes sure this array stays up to date?